### ISSUE TITLE
Tests failing in all builds

### ISSUE THREAD

**Issue by zanchey** [fish-shell/fish-shell#6539]
Since 8a9c39d433fa...38f43306832d on master, all PPA and OBS builds are failing the tests due to the following:

```
Testing Ctrl-C cancellation. If this hangs, that's a bug!
[31mError: Test failed on line 1042: res.status.signal_exited() && res.status.signal_code() == SIGINT[0m
[31mError: Test failed on line 1042: res.status.signal_exited() && res.status.signal_code() == SIGINT[0m
[31mError: Test failed on line 1042: res.status.signal_exited() && res.status.signal_code() == SIGINT[0m
[31mError: Test failed on line 1042: res.status.signal_exited() && res.status.signal_code() == SIGINT[0m
[31mError: Test failed on line 1042: res.status.signal_exited() && res.status.signal_code() == SIGINT[0m
```

I suspect this is something to do with the way the tests are run under package building conditions, that is without a terminal attached (Travis attaches a TTY, whereas pbuilder/rpmbuild do not).

**Comment by zanchey** [comment]
I can reproduce this locally with `ssh localhost 'cd FISH_BUILD_DIR; make test'`.

**Comment by zanchey** [comment]
The process spews a lot of `pthread_create: Cannot allocate memory`, which I suspect is related.

**Comment by ridiculousfish** [comment]
Nice. Also reproes when using lldb. [Backtrace](https://gist.github.com/ridiculousfish/107d124ccc84c8a8c07b4e85cd119647).

**Comment by ridiculousfish** [comment]
Bleh, it's one of those horrible teardown ordering issues.

**Comment by ridiculousfish** [comment]
The underlying issue should only affect the tests because only the tests invoke destructors at exit, but it will mean that 3.1 tests will fail in some cases.

**Comment by zanchey** [comment]
I'm sorry to say this is still happening on all platforms, including macOS, as of 7d486c121.

**Comment by ridiculousfish** [comment]
Yikes! Nice double-catch. I think the lldb-repro must have been a different issue.

**Comment by ridiculousfish** [comment]
I'll bravely close this again with the new fix in af22d6b732fa54dc994f7be9aba364cd11356e63
